java/arcs/tools/Recipe2Plan.kt

```diff
@@ -0,0 +1,57 @@
+package arcs.tools
+
+import arcs.core.storage.api.DriverAndKeyConfigurator
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.arguments.argument
+import com.github.ajalt.clikt.parameters.arguments.multiple
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.flag
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.types.file
+import com.squareup.kotlinpoet.FileSpec
+import java.io.File
+
+/** Generates plans from recipes. */
+class Recipe2Plan : CliktCommand(
+    help = """Generate plans from recipes.
+    
+    This script reads serialized manifests and generates Kotlin files with [Plan] classes.""",
+    printHelpOnEmptyArgs = true
+) {
+    private val outdir by option(help = "output directory; defaults to '.'").file(fileOkay = false)
+    // TODO(b/161994250): Package should be derived from proto
+    private val packageName by option(help = "scope to specified package; default: 'arcs'")
+        .default("arcs")
+    private val manifests by argument(help = "paths to protobuf-serialized manifests")
+        .file(exists = true).multiple()
+    private val verbose by option("--verbose", "-v", help = "Print logs").flag(default = false)
+    // TODO(b/162273478) CLI should accept `policies` argument
+
+    /** Execute: Generate a plan per input manifest proto */
+    override fun run() = manifests.forEach { manifest ->
+        DriverAndKeyConfigurator.configure(null)
+
+        val outputFile = outputFile(manifest)
+        if (verbose) {
+            echo("$manifest --> $outputFile")
+        }
+
+        val fileBuilder = FileSpec.builder(packageName, "")
+
+        fileBuilder.addComment("GENERATED CODE -- DO NOT EDIT")
+
+        // TODO: Generate Plans
+        // val manifestProto = ManifestProto.parseFrom(manifest.readBytes())
+
+        outputFile.writeText(fileBuilder.build().toString())
+    }
+
+    /** Produces a File object per user specification, or with default values. */
+    private fun outputFile(manifest: File): File {
+        val outputName = manifest.name.replace(".pb.bin", ".jvm.kt")
```

Comment:
It would be worth checking somewhere that the file given in `--manifest` actually has the correct file extension.

Reply:
It's checked in the bazel rule rn. However, if we expect people to call the binary directly, it would be good to check here, too.

Reply:
If you go with the `--outfile` param like I suggested, then you can remove this line entirely. That's probably better, then you don't need to rely on the input file having a certain extension; your binary will work correctly with whatever pair of input and output files it is given.